tests/lib/rules/no-get-with-default.js

```diff
@@ -17,25 +17,70 @@ ruleTester.run('no-get-with-default', rule, {
     "getWithDefault.testMethod(testClass, 'key', [])",
   ],
   invalid: [
+    // this.getWithDefault
     {
-      code: "const test = this.getWithDefault('key', []);",
+      code: "const test = this.getWithDefault('key', []);", // With a string property.
+      output: "const test = (this.get('key') === undefined ? [] : this.get('key'));",
       errors: [
         {
           message: ERROR_MESSAGE,
           type: 'CallExpression',
         },
       ],
-      output: null,
     },
     {
-      code: "const test = getWithDefault(this, 'key', []);",
+      code: 'const test = this.getWithDefault(SOME_VARIABLE, []);', // With a variable property.
+      output:
+        'const test = (this.get(SOME_VARIABLE) === undefined ? [] : this.get(SOME_VARIABLE));',
```

Comment:
In this scenario (non-static argument) I think we should avoid the double `this.get` calls 🤔 if at all possible.

For example, with the following input:

```js
let test = this.getWithDefault(SOME_VARIABLE, []);
```

We should emit:

```js
let test = this.get(SOME_VARIABLE);
if (test === undefined) {
  test = [];
}
```

Reply:
In general, I don't think a multi-line autofix is very practical, as `this.getWithDefault(...)` is often used in the middle of lengthy statements/expressions, and so we need an autofix which is an expression that can be placed inside of any other arbitrary statement/expression.

```
// one example from my codebase
return this.getWithDefault('dispute.informationRequests', []).sortBy('requestedAt').get('lastObject');
```

If you still think it's worth adding a special case for the variable declaration case you mentioned, perhaps I could do that as a follow-up.